Add manifest and associated files for org.flatpak.ClassicSnakeTerminal

DonaldKellett opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

***

- W.r.t the minimal set of permissions, my app uses `XDG_CONFIG_HOME` to store per-user high scores but does not use any other extra permissions
- W.r.t. the domain used in the app ID, the source code is hosted on https://github.com/DonaldKellett/csnaketerm and project page at https://donaldkellett.github.io/csnaketerm so I do not own any domain myself. What should I use for my app ID? Is a fallback of `org.flatpak.ClassicSnakeTerminal` acceptable?
- W.r.t submitting additional patches to upstream, I don't think it is necessary since my only modification is adding a shell script which invokes the upstream program with a modified HOME variable.

This is my first time packaging a simple app for Flatpak and I would be honored to have my app officially included in Flathub :smile:

nedrichards commented:
@DonaldKellett that's the part under "or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge)." so for github you'd do either com.github.DonaldKellett.csnaketerm or io.github.DonaldKellett.csnaketerm - either works.

DonaldKellett commented:
Thanks for the quick review, I'll rename the app tomorrow when I get to it :+1:

nedrichards commented on org.flatpak.ClassicSnakeTerminal.yml:
```diff
@@ -0,0 +1,50 @@
+app-id: org.flatpak.ClassicSnakeTerminal
+runtime: org.freedesktop.Platform
+runtime-version: '19.08'
```
does this also work with 20.08? Whilst 19.08 is still within security support the more recent release will be for longer.

nedrichards commented on org.flatpak.ClassicSnakeTerminal.yml:
```diff
@@ -0,0 +1,50 @@
+sdk: org.freedesktop.Sdk
+command: run.sh
+modules:
+  - name: python
```
Could you use the python 2.7 in shared modules? This one here https://github.com/flathub/shared-modules using the same shared build config where possible helps ensure that maximum storage deduplication happens for users.

nedrichards commented on org.flatpak.ClassicSnakeTerminal.yml:
```diff
@@ -0,0 +1,50 @@
+    buildsystem: simple
+    build-commands:
+      - ./configure --prefix=/app
+      - make -j$(nproc)
```
rather than nproc could you use $FLATPAK_BUILDER_N_JOBS 

https://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html there's an array of examples if you need https://github.com/search?q=org%3Aflathub+FLATPAK_BUILDER_N_JOBS&type=code

nedrichards commented:
You're fine not to submit the run.sh upstream, as you say it's not necessary ;-)

DonaldKellett commented:
Thanks for the feedback, I have renamed my branch to `io.github.DonaldKellett.csnaketerm` and will submit a pull request once I have incorporated the changes you suggested.
